nf_core/modules/modules_json.py

```diff
@@ -784,6 +787,25 @@ def get_all_modules(self):
 
         return self.pipeline_modules
 
+    def get_all_components(self, component_type):
+        """
+        Retrieves all pipeline modules/subworkflows that are reported in the modules.json
+
+        Returns:
+            (dict[str, [(str, str)]]): Dictionary indexed with the repo urls, with a
+                                list of tuples (component_dir, components) as values
+        """
+        if self.modules_json is None:
+            self.load()
+        if self.pipeline_components is None:
+            self.pipeline_components = {component_type: {}}
```

Comment:
What is this line doing? I don't see `self.pipeline_components[component_type]` assigned to again in this method.

Reply:
oups, that is a relict from me trying to figure out the levels of object we need here. Removing it in the next commit